### ISSUE TITLE
The description for `require a reset to complete the SET_CERTIFICATE request` in the specification is not clear

### ISSUE THREAD

**Issue by Zhiqiang520** [DMTF/libspdm#2062]
The description for `require a reset to complete the SET_CERTIFICATE request` in the specification is not clear.

It doesn't mention --- **After** the Responder has processed the reset:
if the Requester sends the **same** `SET_CERTIFICATE` that was sent before the reset, what to do then?
if the Requester sends the **different** `SET_CERTIFICATE` request than the request that was sent before the reset, what to do then?

1. Refer to paragraph 675 in [DSP0274_1.2.1](https://www.dmtf.org/sites/default/files/standards/documents/DSP0274_1.2.1.pdf).
`The device might require a reset to complete the SET_CERTIFICATE request, potentially so that the
device can generate AliasCert certificates using lower firmware layers. If the device requires a reset to complete
the SET_CERTIFICATE request, then the device shall respond with an ErrorCode=ResetRequired response.`

2. I think more description for `require a reset to complete the SET_CERTIFICATE request` should be added in the specification  [DSP0274_1.2.1](https://www.dmtf.org/sites/default/files/standards/documents/DSP0274_1.2.1.pdf).
May be just like the paragraph 666 in [DSP0274_1.2.1](https://www.dmtf.org/sites/default/files/standards/documents/DSP0274_1.2.1.pdf) for `requires a reset to complete the GET_CSR request`.
`If the device requires a reset to complete the GET_CSR request, then the device shall respond with an
ErrorCode=ResetRequired response. After the Responder has processed the reset, the Requester sends the same
GET_CSR request that was sent before the reset, which signals to the Responder to send the CSR response. If the
Requester sends a different GET_CSR request than the request that was sent before the reset, then the Responder
shall treat it as a new request or reject the different GET_CSR request with an ErrorCode=UnexpectedRequest
response. On receipt of a new GET_CSR request, the Responder can discard any existing CSR responses.`

**Comment by steven-bellock** [comment]
Unlike `GET_CSR`, the Requester doesn't have to do anything if a Responder requires a reset to complete `SET_CERTIFICATE`. If, after a reset, the Requester sends another `SET_CERTIFICATE` the Responder processes it, regardless if it's the same certificate or a different certificate.

**Comment by jyao1** [comment]
> If, after a reset, the Requester sends another SET_CERTIFICATE the Responder processes it, regardless if it's the same certificate or a different certificate.

Is that in SPDM spec? I am surprised on that.

**Comment by steven-bellock** [comment]
The `ResetRequired` was put in to support DICE. If the leaf certificate is generated by the bootrom then, after a `SET_CERTIFICATE`, the reset will generate a new leaf certificate. The Requester doesn't have to do anything.

**Comment by Zhiqiang520** [comment]
So, for the Scenario--the device requires a reset to complete the `SET_CERTIFICATE` request:
1. Requester send the `SET_CERTIFICATE` request to the device.
2. the device respond with an `ErrorCode=ResetRequired` response, due to require a reset to complete the `SET_CERTIFICATE` request.
3. the device processed the reset and completed the `SET_CERTIFICATE` request.
4. Requester send **another** `SET_CERTIFICATE` request(the same or a different `SET_CERTIFICATE` request) to the device.
5. the device respond with an `ErrorCode=ResetRequired` response, due to require a reset to complete the `SET_CERTIFICATE` request.
6. the device processed the reset and completed the SET_CERTIFICATE request.
7. repeat the step-4 ~ 6
... ...

in the above Scenario, It looks like:
a. The device will **always** respond with an `ErrorCode=ResetRequired` response.
In other words, the Requester will always receive an `ErrorCode=ResetRequired` response, the Requester will **never** receive a Successful `SET_CERTIFICATE_RSP` response message.
b. Then the requester how to know whether the device completed the `SET_CERTIFICATE` request **Successfully**?

**Comment by steven-bellock** [comment]
>Then the requester how to know whether the device completed the SET_CERTIFICATE request Successfully?

Requester can issue a `GET_CERTIFICATE` and compare that the portion it sent to the Responder is the same that's returned from the Responder.

**Comment by jyao1** [comment]
Sending 2nd SET_CERTIFICATE will override the previous CERTIFICATE.

**Comment by Wenxing-hou** [comment]
In current code, `libspdm_get_response_set_certificate` will call `libspdm_write_certificate_to_nvm`.
```
    /* set certificate to NV*/
    result = libspdm_write_certificate_to_nvm(slot_id, cert_chain,
                                              cert_chain_size);
```
In the `libspdm_write_certificate_to_nvm`, the `slot_id` is related to the file_name.
`file_name[strlen(file_name) - 1] = (char)(slot_id+'0');`

When requester send 2nd SET_CERTIFICATE for the same slot_id, the responder will override  the previous CERTIFICATE.

**Comment by jyao1** [comment]
please add unit test.
